packages/colorpickers/src/elements/ColorPicker/index.tsx

```diff
@@ -72,73 +70,99 @@ export interface IColorPickerProps
  */
 export const ColorPicker = forwardRef<HTMLDivElement, IColorPickerProps>(
   ({ color, defaultColor, labels = {}, autofocus, onChange, ...props }, ref) => {
-    const isControlled = color !== null && color !== undefined;
-    const [uncontrolledState, dispatch] = useReducer(reducer, toState(defaultColor));
+    const [state, dispatch] = useReducer(reducer, getInitialState(color || defaultColor));
+    const previousComputedColorRef = useRef<IColor>(state.color);
+    const previousStateColorRef = useRef<IColor>(state.color);
+
+    const controlledColor = useMemo(() => {
+      if (color) {
+        if (typeof color === 'string') {
+          const convertedColor = convertStringToColor(color);
+
+          if (convertedColor) {
+            return convertedColor;
+          }
+        } else {
+          return color;
+        }
+      }
+
+      return state.color;
+    }, [color, state.color]);
+
+    useEffect(() => {
+      if (
+        !areColorsEqual(previousStateColorRef.current, state.color) &&
+        !areColorsEqual(color, state.color)
+      ) {
+        onChange && onChange(state.color);
+      }
+
+      previousStateColorRef.current = state.color;
+    }, [color, onChange, state.color]);
 
     useEffect(() => {
-      if (isControlled === false) {
-        onChange && onChange(uncontrolledState);
+      if (!areColorsEqual(controlledColor, previousComputedColorRef.current)) {
+        dispatch({ type: 'RESET_COLOR', payload: controlledColor });
```

Comment:
Whenever a new controlled value is applied we reset the temporary color and the associated inputs.